src/vector/mod.rs

```diff
@@ -1899,9 +1912,22 @@ impl<'a, A: Clone> DoubleEndedIterator for Iter<'a, A> {
         let focus: &'a mut Focus<'a, A> = unsafe { &mut *(&mut self.focus as *mut _) };
         focus.get(self.back_index)
     }
+
+    fn nth_back(&mut self, n: usize) -> Option<Self::Item> {
+        if n >= self.len() {
+            None
+        } else {
+            self.back_index -= n;
+            self.next_back()
+        }
+    }
 }
 
-impl<'a, A: Clone> ExactSizeIterator for Iter<'a, A> {}
+impl<'a, A: Clone> ExactSizeIterator for Iter<'a, A> {
+    fn len(&self) -> usize {
+        self.back_index - self.front_index
```

Comment:
What if this underflows?
```suggestion
        self.back_index.saturating_sub(self.front_index)
```